pymaker/auctions.py

```diff
@@ -155,19 +159,39 @@ def tick(self, id: int) -> Transact:
 
         return Transact(self, self.web3, self.abi, self.address, self._contract, 'tick', [id])
 
-    def get_past_lognotes(self, number_of_past_blocks: int, abi: list) -> List[LogNote]:
-        assert isinstance(number_of_past_blocks, int)
+    def get_past_lognotes(self, abi: list, from_block: int, to_block: int = None, chunk_size=20000) -> List[LogNote]:
+        current_block = self._contract.web3.eth.blockNumber
+        assert isinstance(from_block, int)
+        assert from_block < current_block
+        if to_block is None:
+            to_block = current_block
+        else:
+            assert isinstance(to_block, int)
+            assert to_block >= from_block
+            assert to_block <= current_block
+        assert chunk_size > 0
         assert isinstance(abi, list)
 
-        block_number = self._contract.web3.eth.blockNumber
-        filter_params = {
-            'address': self.address.address,
-            'fromBlock': max(block_number - number_of_past_blocks, 0),
-            'toBlock': block_number
-        }
+        start = from_block
+        end = None
+        chunks_queried = 0
+        events = []
+        while end is None or start <= to_block:
+            chunks_queried += 1
+            end = min(to_block, start + chunk_size)
+
+            filter_params = {
+                'address': self.address.address,
+                'fromBlock': start,
+                'toBlock': end
+            }
+            logger.debug(f"Querying logs from block {start} to {end} ({end-start} blocks); "
+                         f"accumulated {len(events)} events in {chunks_queried-1} requests")
+
+            logs = self.web3.eth.getLogs(filter_params)
+            events.extend(list(map(lambda l: self.parse_event(l), logs)))
```

Comment:
Could you help me understand what `self.parse_event(l)` is doing? I see `parse_event` is implemented below: 
```
def parse_event(self, event):
     raise NotImplemented()
```
But I'm confused as to what exactly `NotImplemented` is doing in this context..

Reply:
`AuctionContract` is an abstract base class; `parse_event` is implemented by the subclass to interpret the log event for each type of auction.

Reply:
Ah, got it. I now see the `parse_event(...)` in each of its subclasses (`Flipper`, `Flapper`, and `Flopper`).
I should have just looked at the rest of `auctions.py`. Thank you